File: hanoiTower.java
public class Hanoi {
   
   public static void MoveDisk (int from, int to) {
      System.out.printf("%d" -> "%d"\n", from, to);
   }
   
   public static void moveStack (int n, int from, int to) {
      if (n != 0) {
         int other = 6 - (from + to);
         moveStack(n-1, from, other);
         moveDisk(from, to);
         moveStack(n-1, other, to);
      }   
   }
   
   
   public static void main(String[] args) {
      moveStack(3,1,3);
   }
}

File: Hanoi_Tower.py
def move_disk(from_peg, to_peg):
    print(from_peg, '->', to_peg)

def move_stack(num_disks, from_peg, to_peg):
    if num_disks == 1:
        move_disk(from_peg, to_peg)
    else:
        other_peg = 6 - (from_peg + to_peg)
        move_stack(num_disks - 1, from_peg, other_peg)
        move_disk(from_peg, to_peg)
        move_stack(num_disks - 1, other_peg, to_peg)
        
        
